### ISSUE TITLE
mod's handle_node_drops can't handle tsm_surprise

### ISSUE THREAD

**Issue by tacotexmex** [minetest-mods/item_drop#12]
When digging a Surprise block the game crashes: 

```2018-06-17 09:52:37: ACTION[Server]: singleplayer digs tsm_surprise:question at (-465,9,162)
2018-06-17 09:52:37: ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod 'tsm_surprise' in callback node_on_dig(): .../item_drop/init.lua:227: Couldn't spawn item
2018-06-17 09:52:37: ERROR[Main]: stack traceback:
2018-06-17 09:52:37: ERROR[Main]: 	[C]: in function 'error'
2018-06-17 09:52:37: ERROR[Main]: 	.../item_drop/init.lua:227: in function 'handle_node_drops'
2018-06-17 09:52:37: ERROR[Main]: 	...2d/minetest.app/Contents/Resources/builtin/game/item.lua:598: in function <...2d/minetest.app/Contents/Resources/builtin/game/item.lua:538>
```
Any ideas, @HybridDog? Is it perhaps that the code can't handle multiple drops?

**Comment by tacotexmex** [comment]
Speaking of breaking any deployed mod @sofar, would you care to look at the code too? I can't read it 😕

**Comment by tacotexmex** [comment]
Moved all rewrite commits and my recent commits back into a dev branch until this has been resolved.

**Comment by HybridDog** [comment]
It couldn't handle nodes which drop nothing.
Here's the fix (and the more verbose error message):
```Diff
diff --git a/init.lua b/init.lua
index 70b44d2..ee320b6 100644
--- a/init.lua
+++ b/init.lua
@@ -216,10 +216,16 @@ and not minetest.settings:get_bool("creative_mode") then
 				name = item:get_name()
 			end
 
+			if name == "" then
+				-- Sometimes nothing should be dropped
+				count = 0
+			end
+
 			for _ = 1,count do
 				local obj = minetest.add_item(pos, name)
 				if not obj then
-					error("Couldn't spawn item")
+					error("Couldn't spawn item \"" .. name ..
+						"\", drops: " .. dump(drops))
 				end
 
 				local vel = obj:getvelocity()
```

**Comment by tacotexmex** [comment]
Thank you very much. Resolved in 3e6a731caba803c9da0dd4db7c51a3dda8dc653f.
